### PR TITLE
format-date fix with tests

### PR DESCRIPTION
**Pull request by wpride**
Remake of https://github.com/dimagi/commcare-core/pull/377 into master with tests, confirmed fails without the fix. 

@phillipm @ctsims

### CODE REVIEW

File: javarosa/src/test/java/org/javarosa/form/api/test/FormatDateTest.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+ */
+public class FormatDateTest {
+    private FormParseInit fpi;
+
+    /**
+     * load and parse form
+     */
+    @Before
+    public void initForm() {
+        System.out.println("init FormDateTest");
```
Comment (snopoke) [inline javarosa/src/test/java/org/javarosa/form/api/test/FormatDateTest.java]: println

Comment (wpride) [inline javarosa/src/test/java/org/javarosa/form/api/test/FormatDateTest.java]: Think we use sysout for tests but maybe @phillipm can confirm

File: javarosa/src/test/resources/format_date_tests.xml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            <bind nodeset="/data/display_format_wrapped" />
+            <bind nodeset="/data/date_wrapped" calculate="format-date(date(/data/date), '%e %b, %Y')" />
+            <bind nodeset="/data/date_unwrapped" calculate="format-date(/data/date, '%e %b, %Y')" />
+            <itext>
+                <translation lang="en" default="">
+                    <text id="display_format_unwrapped-label">
+                        <value><output value="/data/date_unwrapped" /></value>
+                    </text>
+                    <text id="display_format_wrapped-label">
+                        <value><output value="/data/date_unwrapped" /></value>
```
Comment (phillipm) [inline javarosa/src/test/resources/format_date_tests.xml]: `/data/date_unwrapped` -> `/data/date_wrapped`

File: javarosa/src/test/java/org/javarosa/form/api/test/FormatDateTest.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    public void testAnswerQuestion() {
+        DateData ans;
+        FormEntryController fec = fpi.getFormEntryController();
+        FormEntryPrompt prompt;
+        fec.jumpToIndex(FormIndex.createBeginningOfFormIndex());
+        Localizer l = fpi.getFormDef().getLocalizer();
+        l.setDefaultLocale(l.getAvailableLocales()[0]);
+        l.setLocale(l.getAvailableLocales()[0]);
+
+        do {
```
Comment (phillipm) [inline javarosa/src/test/java/org/javarosa/form/api/test/FormatDateTest.java]: do/while/continue isn't needed if you add one additional `fec.stepToNextEvent()` call

File: javarosa/src/test/java/org/javarosa/form/api/test/FormatDateTest.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                continue;
+            }
+
+            ans = new DateData(new Date());
+            fec.answerQuestion(ans);
+            fec.stepToNextEvent();
+
+            prompt = fpi.getFormEntryModel().getQuestionPrompt();
+            String unwrappedDateString = prompt.getLongText();
+            String javaDateString = new SimpleDateFormat("dd MMM, yyyy").format(new Date());
```
Comment (phillipm) [inline javarosa/src/test/java/org/javarosa/form/api/test/FormatDateTest.java]: `new SimpleDateFormat("dd MMM, yyyy")` -> `new SimpleDateFormat("dd MMM, yyyy", Locale.US)` to get rid of the lint warnings

File: javarosa/src/test/java/org/javarosa/form/api/test/FormatDateTest.java (hunk 1 above)
Comment (phillipm) [inline javarosa/src/test/java/org/javarosa/form/api/test/FormatDateTest.java]: The logging method in tests doesn't really matter, as far as I'm concerned.

**Pull request merged**